build.sh

```diff
@@ -1,46 +1,141 @@
+[ -z "${DOTNET_HOME:-}"] && DOTNET_HOME="$HOME/.dotnet"
+verbose=false
+update=false
+repo_path="$DIR"
 
-buildFolder=".build"
-buildFile="$buildFolder/KoreBuild.sh"
+__usage() {
+    echo "Usage: $(basename ${BASH_SOURCE[0]}) [options] [[--] <MSBUILD_ARG>...]"
+    echo ""
+    echo "Arguments:"
+    echo "    <MSBUILD_ARG>...         Arguments passed to MSBuild. Variable number of arguments allowed."
+    echo ""
+    echo "Options:"
+    echo "    --verbose                Show verbose output."
+    echo "    -c|--channel <CHANNEL>   The channel of KoreBuild to download. Defaults to '$channel'."
+    echo "    -d|--dotnet-home <DIR>   The directory where .NET Core tools will be stored. Defaults to '\$DOTNET_HOME' or '\$HOME/.dotnet."
+    echo "    -p|--path <PATH>         The directory to build. Defaults to the directory containing the script."
+    echo "    -s|--tools-source <URL>  The base url where build tools can be downloaded. Defaults to '$tools_source'."
+    echo "    -u|--update              Update to the latest KoreBuild even if the lock file is present."
+    echo ""
+    echo "Description:"
+    echo "    This function will create a file \$DIR/korebuild-lock.txt. This lock file can be committed to source, but does not have to be."
+    echo "    When the lockfile is not present, KoreBuild will create one using latest available version from \$channel."
+    exit 2
+}
 
-if test ! -d $buildFolder; then
-    echo "Downloading KoreBuild from $koreBuildZip"
+#
+# Functions
+#
 
-    tempFolder="/tmp/KoreBuild-$(uuidgen)"
-    mkdir $tempFolder
+get_korebuild() {
 
-    localZipFile="$tempFolder/korebuild.zip"
+    local lock_file="$repo_path/korebuild-lock.txt"
+    if [ ! -f $lock_file ] || [ "$update" = true ]; then
+        __get_remote_file "$tools_source/korebuild/channels/$channel/latest.txt" $lock_file
+    fi
+    local version=$(cat $lock_file | tail -1 | tr -d '[:space:]')
+    local korebuild_path="$DOTNET_HOME/buildtools/korebuild/$version"
+    if [ ! -d "$korebuild_path" ]; then
+        mkdir -p "$korebuild_path"
+        local remote_path="$tools_source/korebuild/artifacts/$version/korebuild.$version.zip"
 
-    retries=6
-    until (wget -O $localZipFile $koreBuildZip 2>/dev/null || curl -o $localZipFile --location $koreBuildZip 2>/dev/null)
-    do
-        echo "Failed to download '$koreBuildZip'"
-        if [ "$retries" -le 0 ]; then
-            exit 1
+        if ! __machine_has unzip; then
+            echo 'Missing required command: unzip'
+            return 1
         fi
-        retries=$((retries - 1))
-        echo "Waiting 10 seconds before retrying. Retries left: $retries"
-        sleep 10s
-    done
 
-    unzip -q -d $tempFolder $localZipFile
+        tmpfile="$(mktemp)"
+        echo "Downloading KoreBuild $version"
+        __get_remote_file $remote_path $tmpfile
+        unzip -q -d "$korebuild_path" $tmpfile
+        rm $tmpfile || :
```

Comment:
`:` is a builtin equivalent to `true`. It makes this line is functionally equivalent to `rm $tmpfile >/dev/null 2>&1`, except that we'll get some console output indicating there was a problem.

Try this:
```sh
rm some_bogus_file || :
echo $?
```
It should return 0, but still show you that some_bogus_file did not exist.


FYI - https://unix.stackexchange.com/questions/31673/what-purpose-does-the-colon-builtin-serve